### PR TITLE
tests pass in julia 1.0

### PR DESCRIPTION
**Pull request by denizyuret**
Julia 1.0 compatible version.

### CODE REVIEW

**Comment by andreasnoack** [comment]
You'd have to update `.travis.yml` as well. Could you try removing the `script` part and enable testing on 1.0?

**Comment by denizyuret** [comment]
Updated travis.

File: .travis.yml [hunk 1]
```diff
@@ -3,13 +3,11 @@ os:
     - linux
     - osx
 julia:
-    - 0.5
-    - 0.6
+    - 1.0
```
Comment (alyst) [inline .travis.yml]: also 0.7? (as in REQUIRE)

File: src/MAT.jl [hunk 2]
```diff
@@ -143,7 +143,7 @@ end
 Write a dictionary containing variable names as keys and values as values
 to a Matlab file, opening and closing it automatically.
 """
-function matwrite{S, T}(filename::AbstractString, dict::Associative{S, T})
+function matwrite(filename::AbstractString, dict::AbstractDict{S, T}) where {S, T}
```
Comment (alyst) [inline src/MAT.jl]: IIUC type params unnecessary

File: .travis.yml (hunk 1 above)
Comment (denizyuret) [reply .travis.yml]: General recommendation was no 0.7 testing if 1.0 passes.

File: src/MAT.jl (hunk 2 above)
Comment (denizyuret) [reply src/MAT.jl]: cleaned up.

**Comment by alyst** [comment]
Looks like CI fails during HDF5.jl installation. I'm not sure what's the issue, but HDF5.jl has
```yml
addons:
  apt:
    packages:
    - hdf5-tools
```
in its .travis.yml

**Comment by denizyuret** [comment]
I don't think we get an error with the new HDF5 0.10.  My tests pass.  I updated REQUIRE.

**Comment by denizyuret** [comment]
CI failed again. This seems to be a problem with CI building HDF5 not being able to download stuff. Builds and tests ok on a regular machine.  Can we release this?

**Comment by alyst** [comment]
It looks like Julia tries to install `hdf5-tools` during `Pkg.build()`. IMHO it's still worth to try fixing it by updating travis.yml.

**Comment by denizyuret** [comment]
How do you suggest we do that?

**Comment by alyst** [comment]
Just add `hdf5-tools` as "addon" (see the earlier comment, also [HDF5's travis.yml](https://github.com/JuliaIO/HDF5.jl/blob/master/.travis.yml))

**Comment by denizyuret** [comment]
Seems to fail sporadically.

**Comment by denizyuret** [comment]
And I can't make it fail on my machine to debug :(
Can anybody replicate the CI failure?

**Comment by quaertym** [comment]
@denizyuret It is possible to ssh to the travis CI machine.

**Comment by denizyuret** [comment]
@emreunal Can you tell what is causing the issue?

File: test/write.jl [hunk 3]
```diff
@@ -81,7 +81,7 @@ test_write(Dict(
 @test_throws ErrorException test_write(Dict("another invalid key" => "invalid characters"))
 @test_throws ErrorException test_write(Dict("yetanotherinvalidkeyyetanotherinvalidkeyyetanotherinvalidkeyyetanotherinvalidkey" => "too long"))
 
-type TestCompositeKind
+struct TestCompositeKind
```
Comment (woest) [inline test/write.jl]: Shouldn't `type` be converted to `mutable struct`?

Comment (woest) [inline test/write.jl]: Shouldn't `type` be converted to `mutable struct`?

**Pull request closed without merging**